src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/shattered_halls/shattered_halls.cpp

```diff
@@ -107,6 +108,41 @@ void instance_shattered_halls::OnCreatureCreate(Creature* creature)
             m_guidGauntletNPC = creature->GetObjectGuid();
             break;
     }
+    switch (creature->GetGUIDLow())
+    {
+        case HALLOFFATHER_GUID_1:        
+        case HALLOFFATHER_GUID_3:
+        case HALLOFFATHER_GUID_4:
+        case  HALLOFFATHER_GUID_5:
+            creature->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PLAYER);
+            creature->SetVisibility(VISIBILITY_OFF);
+            m_vHallOfFatherGuids.push_back(creature->GetObjectGuid());
+            break;
+        case HALLOFFATHER_GUID_2:
+            creature->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PLAYER);
+            creature->SetVisibility(VISIBILITY_OFF);
```

Comment:
This seems fishy. Explain the purpose to me please.

Reply:
In the Hall of the Fathers we have 3 Groups that have a little Intro RP event.
![gm on](https://user-images.githubusercontent.com/6261245/106888975-9f3d3800-66e7-11eb-9105-25421d0b9b40.jpg)

They are spawned on left/right side and invisible for Players till there Event get triggert.
The First group gets triggert when both of the sentrys are dead.
![ss+(2021-02-04+at+12 45 37)](https://user-images.githubusercontent.com/6261245/106889103-d7dd1180-66e7-11eb-9a1d-6e731c2fe033.jpg)

The 2nd and 3rd group getting trigger when a player is in range of 60y with 2nd group legionnaire.

2 legionnairs using 2 different waypoints one for the intro (creature_movement_template) and one for there normal waypoints. (creature_movement).

Im gonna post a video that show its when i have some time.

Reply:
Well, you could instead use this hot off the press new function:
https://github.com/cmangos/mangos-tbc/blob/master/src/game/Entities/Object.cpp#L2044
which summons a guid from DB on demand. (in db it should have spawnmask 0)

Reply:
Thats actually a nice function. 
But some Questions i already have: 
They get a new guid on spawn (atleast shown ingame via npc info command), does the creature_linking still work?
and how about the legionnaires, we still need the GUID for reinforcment.
and one legionnair still have to be in the map for MoveInLineOfSight to trigger 2nd rp event.  so we have to keep him invisible and UNIT_FLAG_IMMUNE_TO_PLAYER